SIP-0063: Fix Staking Bug to Prevent Reverting Delegated Voting Power (#63)
DistributedCollective/SIPS: SIP-0063.md

---
SIP: '0063'
Title: Fix Staking Bug to Prevent Reverting Delegated Voting Power
Author: Tyrone Johnson (@tjcloa)
Status: Draft
Track: Contract
Created: 2023-05-26
---

# SIP-0063: Fix Staking Bug to Prevent Reverting Delegated Voting Power

## Description  

The Staking contract has been paused to prevent malicious use of the information disclosed by this SIP.

> If approved, this proposal will upgrade the Staking module contract StakingStakeModule to an implementation that fixes the bug.

Reviewer: ```suggestion
If approved, this proposal will upgrade the StakingStakeModule contract to an implementation that fixes the bug.
```

## Details

> A security researcher has reported a bug through Sovryn's Immunefi bug bounty program, which we verified amd confirmed. 
> The bug allows any address to revert delegated voting power back to any delegator's address by staking any amount to the delegator's address.  

Reviewer: ```suggestion
A security researcher has reported a bug through Sovryn's Immunefi bug bounty program. The bug allows any address to revert any delegated voting power back to the delegator's address by staking any amount to the delegator's address. We have reproduced the bug and confirmed this is unintended behavior.
```
